frontend/src/js/components/settings/organization/Billing.test.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import React from 'react';
+
+import * as OrganizationActions from '@northern.tech/store/organizationSlice/thunks';
+import { act, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { defaultState, undefineds } from '../../../../../tests/mockData';
+import { render } from '../../../../../tests/setupTests';
+import { Billing } from './Billing';
+
+const preloadedState = {
+  organization: {
+    ...defaultState.organization,
+    organization: {
+      ...defaultState.organization.organization,
+      billing_profile: {
+        email: 'ok@ok.ok',
+        name: 'Test account',
+        address: {
+          country: 'NO',
+          city: 'test City',
+          line1: 'Test address',
+          postal_code: '1234',
+          state: 'test State'
+        },
+        shipping: {
+          name: 'Test account',
+          address: {
+            country: 'NO',
+            city: 'test City',
+            line1: 'Test address',
+            postal_code: '1234',
+            state: 'test State'
+          }
+        }
+      }
+    },
+    card: {
+      last4: '1234',
+      expiration: { month: 8, year: 1230 },
+      brand: 'Visa'
+    }
+  }
+};
+
+const editProfileActionParams = {
+  billingProfile: {
+    address: {
+      city: 'Oslo',
+      country: 'PL',
+      line1: 'Blindernveien',
+      postal_code: '5678',
+      state: 'Oslo'
+    },
+    email: 'ok@ok.ok',
+    name: 'Test account'
+  }
+};
+
+describe('Billing Component', () => {
+  it('renders correctly', async () => {
+    const { baseElement } = render(<Billing />, {});
+    const view = baseElement;
+    expect(view).toMatchSnapshot();
+    expect(view).toEqual(expect.not.stringMatching(undefineds));
+  });
+
+  it('supports modifying billing profile', async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    const ui = <Billing />;
+    const editProfileAction = jest.spyOn(OrganizationActions, 'editBillingProfile');
+    render(ui, { preloadedState });
+    expect(screen.getByText(/1234, test city/i)).toBeVisible();
```

Comment:
```suggestion
    await expect(screen.getByText(/1234, test city/i)).toBeVisible();
```
to allow the selector to be resolved